core/arch/x86_64/mm/core_mmu.c

```diff
@@ -2487,6 +2544,8 @@ void core_mmu_set_user_map(struct core_mmu_user_map *map)
 {
 	if (map == NULL)
 		clear_user_map();
+	else
+		x86_set_cr3((uint64_t)&g_thread_pml4[thread_get_id()][0]);
```

Comment:
If map is NULL, then clear user map, otherwise, set the mapping table of thread. 
In ARM, this map saved the ttbr0 and asid value, so it will be take effect when setting map. 
But X86, just switch CR3 is enough, don't need these values in the structure of map.